src/main/java/software/amazon/qldb/exceptions/ExecuteException.java

```diff
@@ -21,17 +21,37 @@
  * <p><b>Note</b>: this class is for internal use only.</p>
  */
 public class ExecuteException extends RuntimeException {
-    public SdkException cause;
-    public boolean isRetriable;
-    public boolean isAborted;
-    public boolean isISE;
-    public String txnId;
+    private final SdkException cause;
+    private final boolean retryable;
+    private final boolean aborted;
+    private final boolean ise;
+    private final String txnId;
 
-    public ExecuteException(SdkException cause, boolean isRetriable, boolean isAborted, boolean isISE, String txnId) {
+    public ExecuteException(SdkException cause, boolean isRetryable, boolean isAborted, boolean isISE, String txnId) {
         this.cause = cause;
-        this.isRetriable = isRetriable;
-        this.isAborted = isAborted;
-        this.isISE = isISE;
+        this.retryable = isRetryable;
+        this.aborted = isAborted;
+        this.ise = isISE;
         this.txnId = txnId;
     }
+
+    public SdkException getCause() {
```

Comment:
`getCause` is a method on `Throwable`. You can pass the cause up in the constructor via `super(message, cause)`. See https://docs.oracle.com/javase/7/docs/api/java/lang/Throwable.html#getCause().

Reply:
Calling `super(cause)` will call `RuntimeException(Throwable cause)` constructor. This will set the type of the exception as `Throwable`.

Throwable is a checked exception which means we need to handle it in `driver.execute()` which will be a breaking change.

https://docs.oracle.com/javase/7/docs/api/java/lang/Throwable.html#getCause()
> Throwable and any subclass of Throwable that is not also a subclass of either RuntimeException or Error are regarded as checked exceptions.

Reply:
All exceptions in Java extend Throwable. Calling a super class constructor does now change the type of the class being constructed. This class will continue to extend RuntimeException which will mean that it is not checked.

Reply:
Since we are calling the super class's `getCause()` via inheritance, it is returning type [Throwable](https://docs.oracle.com/javase/7/docs/api/java/lang/Throwable.html#getCause()).

I get an unhandled exception error in my IDE within `QldbDriverImpl.Execute()`
![Screenshot (6)](https://user-images.githubusercontent.com/57543366/111418485-08c84300-86a5-11eb-9158-27714dd8935a.png)

Reply:
Still, overriding getCause with a different implementation is confusing. On second thoughts, you can throw a new exception here (Runtime) with it's cause set to the cause of the ExecutionException, or indeed the ExecutionException itself.

Reply:
Do we know if the underlying exception is checked? If it is, you can optionally wrap it, otherwise throw it directly?